wrappers/python/examples/depth_auto_calibration_example.py

```diff
@@ -1,90 +1,287 @@
 ## License: Apache 2.0. See LICENSE file in root directory.
-## Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+## Copyright(c) 2021 Intel Corporation. All Rights Reserved.
 
 #####################################################
 ##                   auto calibration              ##
 #####################################################
 
-# First import the library
+import argparse
+import json
+import sys
+import time
+
 import pyrealsense2 as rs
-import  sys, getopt
-def read_parameters(argv):
-    opts, args = getopt.getopt(argv, "i:", ["ifile="])
-    jcnt = ''
-    size = 0
 
-    if len(opts) > 0:
-        json_file = opts[0][1]
-        file_object = open(json_file)
-        jcnt = file_object.read()
-        size = len(jcnt)
+__desc__ = """
+This script demonstrates usage of Self-Calibration (UCAL) APIs
+"""
+
+# mappings
+occ_speed_map = {
+    'very_fast': 0,
+    'fast': 1,
+    'medium': 2,
+    'slow': 3,
+    'wall': 4,
+}
+tare_accuracy_map = {
+    'very_high': 0,
+    'high': 1,
+    'medium': 2,
+    'low': 3,
+}
+scan_map = {
+    'intrinsic': 0,
+    'extrinsic': 1,
+}
+fl_adjust_map = {
+    'right_only': 0,
+    'both_sides': 1
+}
+
+ctx = rs.context()
+
+
+def main(arguments=None):
+    args = parse_arguments(arguments)
+
+    try:
+        device = ctx.query_devices()[0]
+    except IndexError:
+        print('Device is not connected')
+        sys.exit(1)
+
+    depth_sensor = device.first_depth_sensor()
+
+    # prepare device
+    depth_sensor.set_option(rs.option.emitter_enabled, 0)
+    if depth_sensor.supports(rs.option.thermal_compensation):
+        depth_sensor.set_option(rs.option.thermal_compensation, 0)
+    if args.exposure == 'auto':
+        depth_sensor.set_option(rs.option.enable_auto_exposure, 1)
+    else:
+        depth_sensor.set_option(rs.option.enable_auto_exposure, 0)
+        depth_sensor.set_option(rs.option.exposure, int(args.exposure))
+
+    print("Starting UCAL...")
+    try:
+        run_on_chip_calibration(args.onchip_speed, args.onchip_scan)
+        run_focal_length_calibration((args.target_width, args.target_height), args.focal_adjustment)
+        run_tare_calibration(args.tare_accuracy, args.tare_scan, args.tare_gt, (args.target_width, args.target_height))
+    finally:
+        if depth_sensor.supports(rs.option.thermal_compensation):
```

Comment:
What is the benefit of turning this on at the end of the example?

Reply:
Resetting camera to its original state